design/vela-core/resourcetracker_design.md

```diff
@@ -0,0 +1,56 @@
+# ResourceTracker: Managing Resources behind Application
+
+- Owner: Da Yin (@somefive)
+- Reviewer: Jian Li (@leejanee), Jianbo Sun (@wonderflow)
+- Date: 12/10/2021
+- Status: Implemented
+
+## Intro
+
+As the release of Workflow in KubeVela v1.1, resources dispatched by application can be extremely dynamic.
+Instead of simply declaring in Application Component, users can have advanced control for applied resources by leveraging WorkflowSteps such as `ApplyObject` or operators such as `op.Delete`.
+Meanwhile, with EnvBinding Policy also released in KubeVela 1.1, users can have multiple same components but deployed in different clusters.
+These techniques raise new challenges for tracking and maintaining of resources.
+
+## Goal
+
+To tackle these challenges, a new architecture of ResourceTracker is proposed and implemented.
+Generally, there are several major technical changes compared to the previous version:
+
+1. Resources do not use OwnerReference to track their ResourceTracker anymore. In other word, the previous bi-directional binding is simplified into uni-directional, which allows us to have more flexible resource management strategies (such as releasing the control of resources).
+2. Resources rendered manifests are recorded in ResourceTracker optionally. Based on that, we can prevent configuration drift by leveraging the reconciling mechanism of the Kubernetes operator pattern. 
+3. ResourceTracker deletion now use finalizer and leverage ApplicationController to reconcile. This ensures the deletion of Application truly removes all managed resources. Also, it allows users to manage versioned resource manually. 
+4. ResourceTracker in the HubCluster can track resources in ManagedClusters, so that no ResourceTracker is needed anymore in ManagedCluster. Now we can use caches for ResourceTracker again. Additionally, we do not individual multicluster garbage-collect logic anymore.
+
+From the perspective of users, the direct new-incoming capabilities include:
+1. Users can prevent configuration drift by default, which is a common usage of the classical Application model. Alternatively, they can only dispatch resources by leveraging [ApplyOnce](../../docs/examples/app-with-policy/apply-once-policy) Policy, which is the mode of Application-as-Workflow.
+2. Users can have customized life-cycle control for application resources by leveraging [GarbageCollect](../../docs/examples/app-with-policy/gc-policy) Policy. For example, users might want to keep resources after version updates or application removal.
+
+## Implementations
+
+### ResourceTracker Types
+
+There are several *ResourceTrackers* maintained for one Application.
```

Comment:
What's the main factor to decide the type of a ResourceTracker?

Reply:
Mainly depending on the life-cycle of resource. For example, if you want to keep a cloud-resource until the application is removed, it will be attached to the ROOT RT. Normally, resources are attached to VERSIONED RTs and will be recycled as soon as they are not in-use.